### ISSUE TITLE
rustfmt: deprecated: rustfmt should now be installed via rustup

### ISSUE THREAD

**Issue by NephIapalucci** [williamboman/mason.nvim#1429]
### I've searched open issues for similar requests

- [X] Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

When installing/updating rustfmt, a warning message is displayed that the installation is deprecated and rustfmt should now be installed via rustup. This issue appeared a couple weeks to a month ago and has persisted since. I assumed this would be fixed/mentioned but I couldn't find any issues for it here.

### Expected behavior

Rustfmt should be installed from the recommended source, or if there is some reason to not do so, a deprecated message should not be shown. 

### Affected packages

rustfmt

### Mason output

```Text
deprecated: rustfmt should now be installed via rustup
```

### Installation log

```Text
[INFO  Fri 28 Jul 2023 12:57:19 AM EDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184: Executing installer for Package(name=rustfmt) {}
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:58: Attempting to lock package Package(name=rustfmt)
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:70: Wrote lockfile Package(name=rustfmt)
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:71: fs: mkdirp /home/neph/.local/share/nvim/mason/staging/rustfmt
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...zy/mason.nvim/lua/mason-core/installer/registry/init.lua:148: Compiling installer. rustfmt {}
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...zy/mason.nvim/lua/mason-core/installer/registry/init.lua:109: Parsing spec rustfmt {}
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...azy/mason.nvim/lua/mason-core/installer/managers/std.lua:95: std: downloading file "https://github.com/rust-lang/rustfmt/releases/download/v1.5.1/rustfmt_linux-x86_64_v1.5.1.tar.gz"
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...ocal/share/nvim/lazy/mason.nvim/lua/mason-core/fetch.lua:37: Fetching URL "https://github.com/rust-lang/rustfmt/releases/download/v1.5.1/rustfmt_linux-x86_64_v1.5.1.tar.gz"
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="curl", spawn_opts={
  args = { "-H", "User-Agent: mason.nvim v1.6.1 (+https://github.com/williamboman/mason.nvim)", "-fsSL", "-X", "GET", "-o", "/home/neph/.local/share/nvim/mason/staging/rustfmt/./rustfmt_linux-x86_64_v1.5.1.tar.gz", "--connect-timeout", 30, "https://github.com/rust-lang/rustfmt/releases/download/v1.5.1/rustfmt_linux-x86_64_v1.5.1.tar.gz" }
}
[DEBUG Fri 28 Jul 2023 12:57:19 AM EDT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 61892
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=61892 exited with exit_code=0, signal=0
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...azy/mason.nvim/lua/mason-core/installer/managers/std.lua:232: std: unpack "rustfmt_linux-x86_64_v1.5.1.tar.gz"
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...azy/mason.nvim/lua/mason-core/installer/managers/std.lua:109: std: untar "rustfmt_linux-x86_64_v1.5.1.tar.gz"
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="tar", spawn_opts={
  args = { "--no-same-owner", "-xvf", "rustfmt_linux-x86_64_v1.5.1.tar.gz" },
  cwd = "/home/neph/.local/share/nvim/mason/staging/rustfmt/."
}
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 61896
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=61896 exited with exit_code=0, signal=0
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:59: fs: unlink /home/neph/.local/share/nvim/mason/staging/rustfmt/./rustfmt_linux-x86_64_v1.5.1.tar.gz
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...zy/mason.nvim/lua/mason-core/installer/registry/link.lua:139: Registering bin links Package(name=rustfmt) {
  rustfmt = "{{source.asset.bin}}"
}
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...zy/mason.nvim/lua/mason-core/installer/registry/link.lua:166: Expanded bin link "rustfmt" -> "rustfmt_linux-x86_64_v1.5.1/rustfmt"
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...vim/lazy/mason.nvim/lua/mason-core/installer/context.lua:229: Promoting cwd "/home/neph/.local/share/nvim/mason/staging/rustfmt" to "/home/neph/.local/share/nvim/mason/packages/rustfmt"
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...nvim/lazy/mason.nvim/lua/mason-core/installer/linker.lua:42: Unlinking Package(name=rustfmt)
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:59: fs: unlink /home/neph/.local/share/nvim/mason/bin/rustfmt
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:47: fs: rmrf /home/neph/.local/share/nvim/mason/packages/rustfmt
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:65: fs: mkdir /home/neph/.local/share/nvim/mason/packages/rustfmt
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:84: fs: rename /home/neph/.local/share/nvim/mason/staging/rustfmt /home/neph/.local/share/nvim/mason/packages/rustfmt
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...nvim/lazy/mason.nvim/lua/mason-core/installer/linker.lua:137: Linking Package(name=rustfmt)
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:39: Building receipt for Package(name=rustfmt)
[DEBUG Fri 28 Jul 2023 12:57:21 AM EDT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:59: fs: unlink /home/neph/.local/share/nvim/mason/staging/rustfmt.lock
[INFO  Fri 28 Jul 2023 12:57:21 AM EDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:245: Installation succeeded for Package(name=rustfmt)
```

### Neovim version (>= 0.7)

NVIM v0.9.1
Build type: Release
LuaJit 2.1.0-beta3

    system vimrc file: "$VIM/sysinit.vim"
    fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info

### Operating system/version

Linux archlinux 6.4.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 11 Jul 2023 05:13:39 +0000 x86_64 GNU/Linux

### Healthcheck

```Text
mason.nvim ~
- OK mason.nvim version v1.6.1
- OK PATH: prepend
- OK Providers: 
  mason.providers.registry-api
  mason.providers.client
- OK neovim version >= 0.7.0

mason.nvim [Registries] ~
- OK Registry `github.com/mason-org/mason-registry version: 2023-07-27-woeful-peony` is installed.

mason.nvim [Core utils] ~
- OK unzip: `UnZip 6.00 of 20 April 2009, by Info-ZIP.  Maintained by C. Spieler.  Send`
- OK wget: `GNU Wget 1.21.4 built on linux-gnu.`
- OK curl: `curl 8.2.0 (x86_64-pc-linux-gnu) libcurl/8.2.0 OpenSSL/3.1.1 zlib/1.2.13 brotli/1.0.9 zstd/1.5.5 libidn2/2.3.4 libpsl/0.21.2 (+libidn2/2.3.4) libssh2/1.11.0 nghttp2/1.55.1`
- OK gzip: `gzip 1.12`
- OK tar: `tar (GNU tar) 1.35`
- OK bash: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
- OK sh: `Ok`

mason.nvim [Languages] ~
- WARNING luarocks: not available
  - ADVICE:
    - spawn: luarocks failed with exit code - and signal -. luarocks is not executable
- WARNING Composer: not available
  - ADVICE:
    - spawn: composer failed with exit code - and signal -. composer is not executable
- WARNING PHP: not available
  - ADVICE:
    - spawn: php failed with exit code - and signal -. php is not executable
- OK Go: `go version go1.20.6 linux/amd64`
- OK Ruby: `ruby 3.0.5p211 (2022-11-24 revision ba5cf0f7c5) [x86_64-linux]`
- WARNING javac: not available
  - ADVICE:
    - spawn: javac failed with exit code - and signal -. javac is not executable
- WARNING java: not available
  - ADVICE:
    - spawn: java failed with exit code - and signal -. java is not executable
- WARNING julia: not available
  - ADVICE:
    - spawn: julia failed with exit code - and signal -. julia is not executable
- OK python: `Python 3.11.3`
- OK node: `v20.4.0`
- OK cargo: `cargo 1.70.0 (ec8a8a0ca 2023-04-25)`
- OK RubyGem: `3.3.25`
- OK npm: `9.8.1`
- OK pip: `pip 23.2.1 from /usr/lib/python3.11/site-packages/pip (python 3.11)`
- OK python venv: `Ok`

mason.nvim [GitHub] ~
- OK GitHub API rate limit. Used: 0. Remaining: 60. Limit: 60. Reset: Fri 28 Jul 2023 01:51:18 AM EDT.
  Install and authenticate via gh-cli to increase rate limit.
```

### Screenshots

![rustfmt](https://github.com/williamboman/mason.nvim/assets/55959655/8f48b43e-1e24-4e73-8400-6699ce23aeb1)

**Comment by williamboman** [comment]
Hello! That's a deprecation message directed at you, the user! rustfmt should no longer be installed via Mason but instead through rustup, due to external factors.

**Comment by NephIapalucci** [comment]
Ah, apologies; I thought Mason was installing rustfmt through some other source which was providing this message to Mason. Is there any particular reason Mason can't do the installation via rustup?

**Comment by williamboman** [comment]
Tl;dr is that Rustup is an entirely self-contained tool and trying to integrate it would violate a bunch of principles Mason relies on, so no unfortunately.
